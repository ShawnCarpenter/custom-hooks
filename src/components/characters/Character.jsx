import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, photoUrl }) => {
  return (
    <figure>
      <img src={photoUrl} alt={name} />
      <figcaption>
        {name}
      </figcaption>
    </figure>
  );
};

Character.propTypes = {
  name:PropTypes.string.isRequired,
  photoUrl:PropTypes.string.isRequired
};

export default Character;
