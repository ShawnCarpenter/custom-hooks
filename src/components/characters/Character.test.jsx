import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(<Character
      name="Analay"
      photoUrl="http://photo.com/img.jpg"/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
