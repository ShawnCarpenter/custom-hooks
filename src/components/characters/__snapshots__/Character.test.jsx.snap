// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Analay"
      src="http://photo.com/img.jpg"
    />
    <figcaption>
      Analay
    </figcaption>
  </figure>
</DocumentFragment>
`;
